File: src/backup.py
import json
import os
from datetime import datetime
from pathlib import Path
from src.db_faq import carregar_faq
from src.logger import logger

BACKUP_DIR = Path("backups")

async def fazer_backup_faq():
    try:
        BACKUP_DIR.mkdir(exist_ok=True)
        itens = await carregar_faq()
        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
        arquivo = BACKUP_DIR / f"faq_backup_{timestamp}.json"

        with open(arquivo, "w", encoding="utf-8") as f:
            json.dump(itens, f, ensure_ascii=False, indent=4)

        # Manter apenas os 5 backups mais recentes
        backups = sorted(BACKUP_DIR.glob("faq_backup_*.json"))
        if len(backups) > 5:
            for antigo in backups[:-5]:
                antigo.unlink()
                logger.info(f"Backup antigo removido: {antigo.name}")

        logger.info(f"Backup do FAQ realizado: {arquivo.name}")
        return True

    except Exception as e:
        logger.error(f"Erro ao fazer backup do FAQ: {e}")
        return False

File: src/core/backup.ts
import { writeFileSync, mkdirSync, readdirSync, unlinkSync } from 'fs';
import path from 'path';
import { faqRepo } from '../db/faqRepo';
import { logger } from './logger';

const BACKUP_DIR = path.join(process.cwd(), 'backups');

export async function fazerBackupFaq(): Promise<boolean> {
  try {
    mkdirSync(BACKUP_DIR, { recursive: true });

    const itens = await faqRepo.carregar();
    const timestamp = new Date()
      .toISOString()
      .replace(/[T:]/g, '_')
      .replace(/\..+/, '');
    const arquivo = path.join(BACKUP_DIR, `faq_backup_${timestamp}.json`);

    writeFileSync(arquivo, JSON.stringify(itens, null, 2), 'utf-8');

    // Manter apenas os 5 backups mais recentes
    const backups = readdirSync(BACKUP_DIR)
      .filter((f) => f.startsWith('faq_backup_') && f.endsWith('.json'))
      .sort();

    if (backups.length > 5) {
      backups.slice(0, -5).forEach((b) => {
        unlinkSync(path.join(BACKUP_DIR, b));
        logger.info({ arquivo: b }, 'Backup antigo removido');
      });
    }

    logger.info({ arquivo: path.basename(arquivo) }, 'Backup do FAQ realizado');
    return true;
  } catch (err) {
    logger.error({ err }, 'Erro ao fazer backup do FAQ');
    return false;
  }
}
